File: cpp/1758.cpp
class Solution {
public:
    int minOperations(string s) {
        auto first = s[0];
        auto second = (first == '1' ?  '0' : '1');
        int result = 0;

        for (int i = 0; i < s.length(); i++) {
            auto c = s[i];
            if (i % 2 == 0 && c != first) result++;
            else if (i % 2 == 1 && c != second) result++;
        }

        int opposite = s.length() - result;
        return (opposite > result ? result : opposite);
    }
};

File: go/1758.go
func minOperations(s string) int {
    first := s[0]
    second := '0'
    if first == '0' {
        second = '1'
    } else {
        second = '0'
    }
    result := 0
    for i,c := range(s) {
        if i % 2 == 0 && c != rune(first) {
            result++
        } else if i % 2 == 1 && c != rune(second) {
            result++
        }
    }
    opposite := len(s) - result
    if result > opposite {
        return opposite
    }
    return result
}
